keylime/web_util.py

```diff
@@ -440,11 +449,11 @@ def validate_api_version(handler, version, logger):
     return True
 
 
-def _list_to_dict(alist):
+def _list_to_dict(alist: List[str]) -> Dict[str, Union[str, None]]:
     """Convert list into dictionary via grouping [k0,v0,k1,v1,...]"""
     params = {}
     i = 0
     while i < len(alist):
-        params[alist[i]] = alist[i + 1] if (i + 1) < len(alist) else None
+        params[alist[i]] = alist[i + 1] if (i + 1) < len(alist) else None  # Must keep NONE here
```

Comment:
It's unfortunate that there is this `None` here. An empty string "" **may be** equivalent an would help to make the return parameter a `Dict[str, str]` but then we do have issues with things like these here (in registrar_common.py for example):

```
        rest_params = web_util.get_restful_params(self.path)

        [...]

        agent_id = rest_params["agents"]

        if agent_id is not None:
```

Reply:
I personally never do a `if var is [not] None`, but `if [not] var`. Maybe this helps here as for one side the code will be equivalent and the type can be forced.

Reply:
I would put it on a todo list because I think this kind of change requires a bit more 'care'.